google-cloud-bigtable/src/main/java/com/google/cloud/bigtable/gaxx/retrying/ApiResultRetryAlgorithm.java

```diff
@@ -43,11 +72,30 @@ public boolean shouldRetry(
     if (context.getRetryableCodes() != null) {
       // Ignore the isRetryable() value of the throwable if the RetryingContext has a specific list
       // of codes that should be retried.
-      return (previousThrowable instanceof ApiException)
-          && context
-              .getRetryableCodes()
-              .contains(((ApiException) previousThrowable).getStatusCode().getCode());
+      return extractRetryDelay(previousThrowable) != null
+          || ((previousThrowable instanceof ApiException)
+              && context
+                  .getRetryableCodes()
+                  .contains(((ApiException) previousThrowable).getStatusCode().getCode()));
     }
     return shouldRetry(previousThrowable, previousResponse);
   }
+
+  public static Duration extractRetryDelay(Throwable throwable) {
```

Comment:
I see its needed for MutateRows.
So given the current design of gax, if this were to be upstreamed, this would be an isntance method on ApiException. However I understand that you cant fork ApiException. So as a compromise I think it would be better to create a class `ApiExceptions` and add the method there, with a comment saying that this belongs on ApiException.

Also, I dont think you want to expose the actual delay to veneer code, but only if the error is retryable. So maybe something like this:

ApiExceptions {
  // TODO: this should replace the existing ApiException#isRetryable() method, but that cant be done in bigtable, so this lives here for now
  static boolean isRetryable2(ApiException e) {
     if (extractRetryDelay(e) != null) {
        return true;
     }
     return e.isRetryable();
}